lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php

```diff
@@ -437,8 +437,14 @@ private function getTableDiffForAlterForeignKey(ForeignKeyConstraint $foreignKey
 
     private function parseColumnCollationFromSQL(string $column, string $sql) : ?string
     {
-        $pattern = '{(?:\W' . preg_quote($column) . '\W|\W' . preg_quote($this->_platform->quoteSingleIdentifier($column))
-                 . '\W)[^,(]+(?:\([^()]+\)[^,]*)?(?:(?:DEFAULT|CHECK)\s*(?:\(.*?\))?[^,]*)*COLLATE\s+["\']?([^\s,"\')]+)}isx';
+        if (PHP_VERSION_ID < 70300) {
+            $pattern = '{(?:\W' . addcslashes(preg_quote($column, '/'), '#') . '\W|\W' . addcslashes(preg_quote($this->_platform->quoteSingleIdentifier($column), '/'), '#')
```

Comment:
Since this pattern changed delimiter from `/` to `{...}`, it is now superfluous to specify second parameter to `preg_quote()`.